versioning/lockfiles/build_order.rst

```diff
@@ -0,0 +1,211 @@
+
+    # Build app1 and dependencies
+    $ conan install app1/0.1@user/testing --build=missing
+
+
+Now that there are binaries for all packages in the cache, capture a new lockfile, and compute its build order:
+
+.. code-block:: bash
+
+    # Capture a new lockfile, which will find all package binaries and fully lock them
+    $ conan lock create --reference=app1/0.1@user/testing --lockfile-out=app1.lock
+    # And check whats need to be built
+    $ conan lock build-order app1.lock --json=build_order.json
+    # The build order is emtpy, nothing to build
+    []
+
+The result of this build order is empty. As the ``lock create`` managed to find existing binaries,
+everything is fully locked, nothing needs to be built.
+
+If we specify ``--build``, then the behavior is different:
+
+.. code-block:: bash
+
+    $ conan lock create --reference=app1/0.1@user/testing --lockfile-out=app1.lock --build
+    # the lockfile will not lock the binaries
+    # And check whats need to be built
+    $ conan lock build-order app1.lock --json=build_order.json
+    # The build order is emtpy, nothing to build
+    [[["pkga/0.1@user/testing", "5ab8...1ac9", "host", "4"]], ...
+
+
+This feature is powerful when combined with ``package_id_modes``, because it can
+automatically define the minimum set of packages that needs to be built for any
+change in the dependency graph.
+
+Let's say that a new version ``pkgb/2.0@user/testing`` is created. But if we
+check the ``pkgd`` *conanfile.py* requirement, we can see that this falls outside
+of the valid version range. Then, it does not affect ``pkgd`` or ``app1`` and
+nothing needs to be built:
+
+.. code-block:: bash
+
+    $ conan create pkgb pkgb/2.0@user/testing
+    $ conan lock create --reference=app1/0.1@user/testing --lockfile-out=app1.lock
+    $ conan lock build-order app1.lock --json=build_order.json
+    [] # Empty, nothing to build, pkgb/2.0 does not become part of app1
+
+
+If on the contrary, a new ``pkgb/0.2@user/testing`` is created, and we capture a
+new lockfile, it will contain such new version. Other packages, like ``pkga`` and
+``pkgc`` are not affected by this new version, and will be fully locked in the lockfile,
+but the dependents of ``pkgb`` now won't be locked and it will be necessary to build them:
+
+.. code-block:: bash
+
+    $ conan create pkgb pkgb/0.2@user/testing
+    $ conan lock create --reference=app1/0.1@user/testing --lockfile-out=app1.lock
+    $ conan lock build-order app1.lock --json=build_order.json
+    [[['pkgd/0.1@user/testing', '97e9...b7f4', 'host', '2']],
+     [['app1/0.1@user/testing', '2bf1...e405', 'host', '1']]]
+
+So in this case the *app1.lock* is doing these things:
+
+- Fully locking the non-affected packages (``pkga/0.1``, ``pkgc/0.1``)
+- Fully locking the ``pkgb/0.2``, as the binary that was just created is valid for our ``app1``.
+  Note that this might not always be true, and ``app1`` build could require a diffeent ``pkgb/0.2``
+  binary.
```

Comment:
```suggestion
- Fully locking the ``pkgb/0.2``, as the binary that was just created is valid for our ``app1``
  (Note that this might not always be true, and ``app1`` build could require a different ``pkgb/0.2``
  binary).
```